### PR TITLE
Design Picker: Switch to `react-intersection-observer` (v2)

### PR DESCRIPTION
**Pull request by danielbachhuber**
See https://github.com/Automattic/wp-calypso/issues/74022
Previously https://github.com/Automattic/wp-calypso/pull/71064 (and https://github.com/Automattic/wp-calypso/pull/75594)

## Proposed Changes

Switches to `react-intersection-observer` in the Design Picker because `calypso/lib/use-in-view` is buggy and it doesn't make sense to spend time reinventing the wheel.

## Testing Instructions

1. Create a new Premium site.
2. Hit `Continue` until you get to the Design Picker.
3. Click on a category and verify `calypso_signup_design_scrolled_to_end` is fired when you hit the end of the page.

image

### CODE REVIEW

File: packages/design-picker/src/components/unified-design-picker.tsx [hunk 1]
```diff
@@ -348,8 +348,20 @@ const UnifiedDesignPicker: React.FC< UnifiedDesignPickerProps > = ( {
...
-	// Track as if user has scrolled to bottom of the design picker
-	const ref = useInView< HTMLDivElement >( onViewAllDesigns, {}, [ categorization?.selection ] );
+	const [ viewedSelections, setViewedSelections ] = useState< string[] >( [] );
+
+	const { ref } = useInView( {
+		onChange: ( inView ) => {
+			if (
+				inView &&
+				categorization?.selection &&
+				! viewedSelections.includes( categorization?.selection )
```
Comment (okmttdhr) [inline packages/design-picker/src/components/unified-design-picker.tsx]: badge badge

This condition will likely prevent the event from firing if the user scrolls to the end a second time in the same category.

e.g., 

- Select `Blog`
- Scroll to the bottom 
	- The event fire 🔥 
- Scroll to the top and Select `Business`
- Scroll to the bottom 
	- The event fire 🔥 
- **Scroll to the top and Select `Blog` again**
- Scroll to the bottom 
	- **The event does not fire** 🧑‍🚒 

I thought this was not intended and should be changed to the original behavior, but WDYT? @Automattic/lego @autumnfjeld

Comment (danielbachhuber) [reply packages/design-picker/src/components/unified-design-picker.tsx]: > This condition will likely prevent the event from firing if the user scrolls to the end a second time in the same category.

This is correct. I thought this was the original intent...

Comment (taipeicoder) [reply packages/design-picker/src/components/unified-design-picker.tsx]: @danielbachhuber thank you for working on this 🙂
I went to the production environment to double-check, and confirmed that the event would trigger again after switching categories (which means that there is no need to keep track of the viewed categories).

**Comment by autumnfjeld** [comment]
> Create a new Premium site.
@danielbachhuber  Why does the site have to be premium? 

The event calypso_signup_design_scrolled_to_end should fire the same way regardless of plan, ya?

I've never looked at data from this event before, but I guess it would make sense for it to behave as @okmttdhr  [describes](https://github.com/Automattic/wp-calypso/pull/76136#discussion_r1176078488), per category.    
Select Blog
Scroll to the bottom
The event fire 🔥
Scroll to the top and Select Business
Scroll to the bottom
The event fire 🔥
Scroll to the top and Select Blog again
Scroll to the bottom
The event does not fire 🧑‍🚒

Also we know for instance such an event is only useful for some categories where a scroll action is needed to see all designs.  For example there is only one podcast, so the event is not providing interesting data there.   In fact, why is it even firing on the podcast page?   Ohhhh, because of the Design your own CTA makes the page scrollable. Hmmm.  

It appears the event was already tracking "per category" (what @taipeicoder  is saying above).

image

At any rate, thanks for making it fire correctly! : )

**Comment by danielbachhuber** [comment]
@autumnfjeld @taipeicoder I'm a little bit confused about this feedback. Should I:
1. Keep the existing behavior and only fire the event once per category.
2. Remove the category tracking and fire the event every time the user hits the bottom of the page.

Thanks!

**Comment by taipeicoder** [comment]
> @autumnfjeld @taipeicoder I'm a little bit confused about this feedback. Should I:
> 
> 1. Keep the existing behavior and only fire the event once per category.
> 2. Remove the category tracking and fire the event every time the user hits the bottom of the page.
> 
> Thanks!

Hi @danielbachhuber, let's keep the existing production behavior (i.e.: not tracking viewed categories) for now. We can change it afterward if needed. Thank you!

**Comment by danielbachhuber** [comment]
> let's keep the existing production behavior (i.e.: not tracking viewed categories) for now. We can change it afterward if needed.

@taipeicoder Sounds good, I've changed the behavior with 0265de98c6

**Pull request merged**